R/edit-spreadsheets.R

```diff
@@ -0,0 +1,173 @@
+    key <- ssfeed_df$sheet_key[key_index]
+  }
+  
+  url <- "https://www.googleapis.com/drive/v2/files"
+  
+  the_url <- slaste(url, key, "trash")
+  
+  gsheets_POST(the_url, the_body = NULL)
+  
+  message('Spreadsheet moved to trash in Google Drive.')
+}
+
+
+#' Make a copy of an existing spreadsheet
+#' 
+#' You can either make a copy of your own spreadsheet or another user's 
+#' spreadsheet. If the spreadsheet you want to make a copy of already exists in 
+#' your Google Drive, pass in the title of the spreadsheet. To get a copy of 
+#' another user's spreadsheet, enter the key of that spreadsheet. Make sure 
+#' that the target spreadsheet is made 'accessible' in the sharing dialog 
+#' options or else it wont be found.
+#' 
+#' @param x the title or key or url of a spreadsheet
+#' 
+#' @param new_title a character string for the new title of the spreadsheet, 
+#' if \code{new_title} is NULL then the copied spreadsheet will
+#' be given the default name: "Copy of ..."
+#' 
+#' @note if two spreadsheets with the same name exist in your Google drive then 
+#' spreadsheet with the most recent "last updated" timestamp will be copied. 
+#' @export
+copy_ss <- function(x, new_title = NULL)
+{
+  # is it an old style sheet?
+  url_start <- "https://docs.google.com/spreadsheet/ccc?key="
+  if(x %>% stringr::str_detect(stringr::fixed(url_start))) {
+    key <- x %>% stringr::str_replace(., stringr::fixed(url_start), "") %>%
+      stringr::str_split_fixed('&', n = 2) %>% "["(1)
+  } else {
+    # is it a new style sheet?
+    url_start <- "https://docs.google.com/spreadsheets/d/"
+    if(x %>% stringr::str_detect(stringr::fixed(url_start))) {
+      key <- x %>% stringr::str_replace(url_start, '') %>%
+        stringr::str_split_fixed('/', n = 2) %>%
+        "["(1)
+    } else {
+      # check user's spreadsheets
+      ssfeed_df <- list_spreadsheets()
+      
+      if(x %in% ssfeed_df$sheet_key) {
+        key <- x
+      } else {
+        key_index <- match(x, ssfeed_df$sheet_title)
+        key <- ssfeed_df$sheet_key[key_index]
+      }
+    }
+  }
+  
+  the_body <- data.frame("title" = new_title) %>% jsonlite::toJSON(.) %>% 
+    stringr::str_sub(., 2, -2)
+  
+  the_url <- slaste("https://www.googleapis.com/drive/v2/files", key, "copy")
+  
+  gsheets_POST(the_url, the_body)
+  
+  message("A copy of the spreadsheet has been made in your Google Drive.")
+}
+
+
+#' Add a new (empty) worksheet to spreadsheet
+#'
+#' Add a new (empty) worksheet to spreadsheet, specify title, worksheet extent (number of rows 
+#' and columns). The title of the new worksheet can not be the same as any 
+#' existing worksheets in the spreadsheet.
+#'
+#' @param ss a registered Google spreadsheet
+#' @param ws_title character string for title of new worksheet 
+#' @param nrow number of rows (default is 1000)
+#' @param ncol number of columns (default is 26)
+#' 
+#' @export
+new_ws <- function(ss, ws_title, nrow = 1000, ncol = 26) 
```

Comment:
Do we have to specify these defaults ourselves? Or can we just say nothing, i.e. not pass extent, and Google will impose its own default extent (which is of course 1000 x 26, at the moment)?